### PR TITLE
[MJLINK-84] Remove remaining commons-lang3

### PR DESCRIPTION
**Pull request by matriv**
Dependency to commons-lang3 has been removed earlier on, but still there were usages of `StringUtils` & `SystemUtils` which lead to the plugin requiring the dependency to be added explicitly when used.

Follows: https://github.com/apache/maven-jlink-plugin/pull/167

Following this checklist to help us incorporate your 
contribution quickly and easily:

 - [X] Make sure there is a [JIRA issue](https://issues.apache.org/jira/browse/MJLINK) filed 
       for the change (usually before you start working on it).  Trivial changes like typos do not 
       require a JIRA issue.  Your pull request should address just this issue, without 
       pulling in other changes.
 - [X] Each commit in the pull request should have a meaningful subject line and body.
 - [X] Format the pull request title like `[MJLINK-XXX] - Fixes bug in ApproximateQuantiles`,
       where you replace `MJLINK-XXX` with the appropriate JIRA issue. Best practice
       is to use the JIRA issue title in the pull request title and in the first line of the 
       commit message.
 - [X] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
 - [X] Run `mvn clean verify` to make sure basic checks pass. A more thorough check will 
       be performed on your pull request automatically.
 - [x] You have run the integration tests successfully (`mvn -Prun-its clean verify`).

If your pull request is about ~20 lines of code you don't need to sign an
[Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf) if you are unsure
please ask on the developers list.

To make clear that you license your contribution under 
the [Apache License Version 2.0, January 2004](http://www.apache.org/licenses/LICENSE-2.0)
you have to acknowledge this by using the following check-box.

 - [x] I hereby declare this contribution to be licenced under the [Apache License Version 2.0, January 2004](http://www.apache.org/licenses/LICENSE-2.0)

 - [ ] In any other case, please file an [Apache Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

### CODE REVIEW

**Review by elharo — CHANGES_REQUESTED** [review]

File: src/main/java/org/apache/maven/plugins/jlink/AbstractJLinkToolchainExecutor.java [hunk 1]
```diff
@@ -193,4 +190,18 @@ private int executeCommand(Commandline cmd) throws MojoExecutionException {
             throw new MojoExecutionException("Unable to execute jlink command: " + e.getMessage(), e);
         }
     }
+
+    private static boolean isOSWindows() {
+        String osName;
```
[the remark this answers is not in the document]
Comment (matriv) [reply src/main/java/org/apache/maven/plugins/jlink/AbstractJLinkToolchainExecutor.java:195]: what do you mean? it's a private method used in two different other places (methods).

File: src/main/java/org/apache/maven/plugins/jlink/AbstractJLinkToolchainExecutor.java [hunk 2]
```diff
@@ -172,7 +169,7 @@ private int executeCommand(Commandline cmd) throws MojoExecutionException {
 
                 StringBuilder msg = new StringBuilder("\nExit code: ");
                 msg.append(exitCode);
-                if (StringUtils.isNotEmpty(err.getOutput())) {
+                if (!err.getOutput().isBlank()) {
```
[the remark this answers is not in the document]
Comment (matriv) [reply src/main/java/org/apache/maven/plugins/jlink/AbstractJLinkToolchainExecutor.java]: addressed, thx.

File: src/main/java/org/apache/maven/plugins/jlink/AbstractJLinkToolchainExecutor.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (matriv) [reply src/main/java/org/apache/maven/plugins/jlink/AbstractJLinkToolchainExecutor.java]: done, thx!

**Review by elharo — APPROVED** [review]

**Review by elharo — CHANGES_REQUESTED** [review]
Need to remove commons-lang3 from pom.xml

Error:  Unused declared dependencies found:
Error:     org.apache.commons:commons-lang3:jar:3.8.1:compile

**Comment by matriv** [comment]
> Need to remove commons-lang3 from pom.xml
> 
> Error: Unused declared dependencies found: Error: org.apache.commons:commons-lang3:jar:3.8.1:compile

apologies, done!

**Review by elharo — APPROVED** [review]

**Pull request merged**